### PR TITLE
Johnstegeman Colemak layout for Keebio Iris

### PR DESCRIPTION
**Pull request by johnstegeman**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keebio/iris/keymaps/johnstegeman/README.md
Comment (waffle87) [inline keyboards/keebio/iris/keymaps/johnstegeman/README.md:1]: This filename should be lowercase.

Comment (johnstegeman) [reply keyboards/keebio/iris/keymaps/johnstegeman/README.md:1]: fixed

File: keyboards/keebio/iris/keymaps/johnstegeman/keymap.c [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+#include "light_layers.h"
+
+enum layers {
+_COLEMAK=0,
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/johnstegeman/keymap.c:22]: The zero isn't needed, as the enum starts at zero, by default. 
```suggestion
_COLEMAK,
```

File: keyboards/keebio/iris/keymaps/johnstegeman/keymap.c [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+
+  switch (keycode) {
+    case COLEMAK:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_COLEMAK);
+      }
+      return false;
+      break;
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/johnstegeman/keymap.c:116]: If this is the only default layer, then this isn't needed and can be removed.

File: keyboards/keebio/iris/keymaps/johnstegeman/keymap.c [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    case ADJUST:
+      if (record->event.pressed) {
+        layer_on(_ADJUST);
+      } else {
+        layer_off(_ADJUST);
+      }
+      return false;
+      break;
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/johnstegeman/keymap.c:144]: Just a heads up, these are built into core:
https://docs.qmk.fm/#/feature_tri_layer

And the adjust layer is basically just `MO(_ADJUST)`

File: keyboards/keebio/iris/keymaps/johnstegeman/keymap.c [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+   case SS_P:
+      if (record->event.pressed) {
+         SEND_STRING("Colemak Rocks!\n");
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/johnstegeman/keymap.c:147]: lol, nice. ;)

File: keyboards/keebio/iris/keymaps/johnstegeman/rules.mk [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ENCODER_ENABLE = no
+#RGB_MATRIX_ENABLE = yes
+#RGB_MATRIX_DRIVER = WS2812
+
+LTO_ENABLE = yes
+VIA_ENABLE = yes
+
+
+RGBLIGHT_ENABLE = yes # by default this is no, but for the rgb light layers to work it needs to be `yes`
+RGB_MATRIX_ENABLE = no # by default this is yes, but for the rgb light layers to work it needs to be `no`
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/johnstegeman/rules.mk:23]: You can do the same with rgb matrix, and the indicators function: 
https://docs.qmk.fm/#/feature_rgb_matrix?id=callbacks

It's a bit more manual, but just as flexible

File: keyboards/keebio/iris/keymaps/johnstegeman/rules.mk [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+SPLIT_KEYBOARD = yes
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/johnstegeman/rules.mk:13]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**